matlab_simulation/05-estimation/kalman_filter/kf_example2.m

```diff
@@ -0,0 +1,109 @@
+% Kalman Filter Example 2: 2D Omnidirectional AUV example
+% 'Estimation 1' topic slide 77
+% State : position and velocity in north, east directions
+% Motion : motion in north and east directions
+% Measurement : measure position in north and east directions
+% Controller : Thrust in north and east directions
+
+clear all
+clc
+close all
+
+% Example number
+example = 2;
+
+% Create AVI object
+makemovie = 1;
+if(makemovie)
+    vidObj = VideoWriter('example2kf.avi');
+    vidObj.Quality = 100;
+    vidObj.FrameRate = 8;
+    open(vidObj);
+end
+
+% Discrete time step
+dt = 0.1;
+
+% Prior
+mu = zeros(4,1); % mean (mu)
```

Comment:
I would recommend declaring a variable (nStates perhaps) and setting it equal to four. This makes the code easier to understand.